### PR TITLE
Move maestro dependencies into the coppersmith API

### PR DESCRIPTION
**Pull request by ewanv**

### CODE REVIEW

File: core/src/main/scala/commbank/coppersmith/api/package.scala [hunk 1]
```diff
@@ -29,6 +32,16 @@ package object api {
   implicit def fromCFS[S, C: TypeTag](fs: ContextFeatureSource[S, C, _]) =
     commbank.coppersmith.FeatureBuilderSource.fromCFS(fs)
 
+  // Mirroring https://github.com/CommBank/maestro/blob/master/maestro-macros/src/main/scala/au/com/cba/omnia/maestro/macros/Macros.scala
+  implicit def derivedEncode[A <: ThriftStruct]: au.com.cba.omnia.maestro.core.codec.Encode[A] =
```
Comment (stephanh) [inline core/src/main/scala/commbank/coppersmith/api/package.scala]: We usually tend to separate types and implicits when generating API packages and often prefer not to put implicits into package objects and put them into a seperate object e.g `Maestro` or `Coppersmith`. This allows users to do a wildcard import on all these types without pulling in implicits that can have much more significant imapcts.

If you create a Copersmith object you could probably just extend MacroSupport and don't even need to specify the implicits again.

File: core/src/main/scala/commbank/coppersmith/api/package.scala [hunk 2]
```diff
@@ -14,20 +14,25 @@
 
 package commbank.coppersmith
 
+import scala.reflect.runtime.universe.TypeTag
+
+import com.twitter.scrooge.ThriftStruct
```
Comment (dkristian) [inline core/src/main/scala/commbank/coppersmith/api/package.scala]: Unused?

File: examples/src/main/scala/commbank/coppersmith/examples/Example2.scala [hunk 3]
```diff
@@ -15,12 +15,10 @@
 package commbank.coppersmith.examples
 
 import commbank.coppersmith.Feature._
-import commbank.coppersmith._
+import commbank.coppersmith.Patterns
```
Comment (dkristian) [inline examples/src/main/scala/commbank/coppersmith/examples/Example2.scala]: This probably implies that it belongs in `api._`.

File: examples/src/main/scala/commbank/coppersmith/examples/Example2.scala [hunk 4]
```diff
@@ -15,12 +15,10 @@
 package commbank.coppersmith.examples
 
 import commbank.coppersmith.Feature._
```
Comment (dkristian) [inline examples/src/main/scala/commbank/coppersmith/examples/Example2.scala]: This import can be removed if `Type.Continuous` is changed to `Continuous` below.

File: examples/src/main/scala/commbank/coppersmith/examples/Example2.scala [hunk 5]
```diff
@@ -15,12 +15,10 @@
...
+import commbank.coppersmith.Patterns
+import commbank.coppersmith.api._, Coppersmith._
 import commbank.coppersmith.examples.thrift._
 
-import au.com.cba.omnia.maestro.scalding.JobStatus
-import au.com.cba.omnia.maestro.api._, Maestro._
-
 import com.twitter.scalding.{TypedPsv, Config, MultipleTextLineFiles, Execution}
 import commbank.coppersmith.Feature.Value.Decimal
 import commbank.coppersmith.scalding.lift.ScaldingScalazInstances
```
Comment (dkristian) [inline examples/src/main/scala/commbank/coppersmith/examples/Example2.scala]: If making the above changes (re `Patterns` and `Continuous`), the import list should be able to be reduced to
```scala
import com.twitter.scalding.{TypedPsv, Config, MultipleTextLineFiles, Execution}

import scalaz.Scalaz._

import org.joda.time.DateTime

import commbank.coppersmith.api._, Coppersmith._
import commbank.coppersmith.scalding.lift.scalding._

import commbank.coppersmith.examples.thrift._
```

There's some vertical alignment and long-line formatting in the code that could be improved while in the area too.

**Comment by dkristian** [comment]
Couple of points around imports and some formatting that could be improved, otherwise :+1:

File: examples/src/main/scala/commbank/coppersmith/examples/Example2.scala (hunk 3 above)
Comment (ewanv) [inline examples/src/main/scala/commbank/coppersmith/examples/Example2.scala]: Should I make that move in this PR, or raise another PR for that?

Comment (dkristian) [inline examples/src/main/scala/commbank/coppersmith/examples/Example2.scala]: I'd piggy back onto this PR, as it's trivial enough and related to rounding out the API in general.

**Pull request merged**